app/javascript/controllers/selection_controller.js

```diff
@@ -0,0 +1,58 @@
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller {
+  static targets = ["rowSelection"];
+  static values = {
+    storageKey: {
+      type: String,
+      default: location.protocol + "//" + location.host + location.pathname,
+    },
+  };
+
+  connect() {
+    this.element.setAttribute("data-controller-connected", "true");
+
+    const storageValue = JSON.parse(
+      sessionStorage.getItem(this.storageKeyValue)
+    );
+
+    if (storageValue) {
+      this.rowSelectionTargets.map((row) => {
+        if (storageValue.indexOf(row.value) > -1) {
+          row.checked = true;
+        }
+      });
+    } else {
```

Comment:
**nitpick (blocking):** There is actually no need to loop through the row targets and see if they are selected since the `connect` method only get run when the stimulus controller connects to the page.  You will need to set an empty array into storage though if one does not exist.

Reply:
Fair enough. Please see d0afd614ed19b17be12e69e03229978532a8ae5c.

Reply:
Thanks